internal/guest/oom/oom.go

```diff
@@ -0,0 +1,78 @@
+// Package oom providers wrappers and support to leverage the OOM watcher functionality
+// in github.com/containerd/containerd/pkg/oom
+package oom
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/Microsoft/hcsshim/internal/guest/prot"
+	log "github.com/Microsoft/hcsshim/internal/log"
+	eventstypes "github.com/containerd/containerd/api/events"
+	"github.com/containerd/containerd/events"
+	"github.com/containerd/containerd/runtime"
+	"github.com/containerd/containerd/runtime/v2/shim"
+	"github.com/pkg/errors"
+)
+
+const emptyActivityID = "00000000-0000-0000-0000-000000000000"
+
+// wraps `bridge.PublishNotification` to conform to `shim.Publish`, so that
+// it can be used by containerd's `oom.Watcher`
+type BridgePublisherFunc func(*prot.ContainerNotification)
```

Comment:
I think maybe we should move all of this to a different package since it's not tied to oom

Reply:
Maybe we could do `internal/guest/bridge/publisher` or `internal/guest/bridgepublisher`

Reply:
done